plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GroovyExtension.java

```diff
@@ -73,16 +75,16 @@ public void importOrder(List<String> importOrder) {
 						"'importOrder([x, y, z])' is deprecated.",
 						"Use 'importOrder x, y, z' instead.",
 						"For details see https://github.com/diffplug/spotless/tree/master/plugin-gradle#applying-to-java-source"));
-		addStep(ImportOrderStep.createFromOrder(importOrder));
+		importOrder(importOrder.toArray(new String[0]));
 	}
 
 	public void importOrder(String... importOrder) {
-		addStep(ImportOrderStep.createFromOrder(importOrder));
+		addStep(importOrderStepFactory.createFrom(importOrder));
```

Comment:
```java
class GroovyExtension {
  public void importOrderFile(String... importOrder) {
    addStep(ImportOrderStep.noSemicolon().createFrom(importOrder));
  }
```

Instance fields are state, and less state is better than more state.

Reply:
Sorry @nedtwigg , I would prefer to keep #218 out, since the purpose is different. Furthermore this PR has only effect on `plugin-gradle` (well, it should :wink: ), whereas  #218 should be only documented in the `CHANGES.md` of `spotless`.